### PR TITLE
Recipe logic errors

### PR DESCRIPTION
**Pull request by huilinn**

### CODE REVIEW

File: recipes/recipe-logic-errors.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
+---
+title: Recipe logic errors
+date: 2018-01-24 15:00:00 Z
+---
+
+# Recipe logic errors
+Even when your recipe doesn't encounter design-time errors that prevent it from running, or execution errors that causes job failures, there might be other recipe logic issues. Some of these issues might be caught during testing, as data isn't moved across apps properly. Other issues might not be caught until the recipe is running. We recommend monitoring your data and recipes closely during testing and the first couple of days after the recipe has been started.
```
Comment (zweedao) [inline recipes/recipe-logic-errors.md]: Grammar: ....or execution errors that **cause** job failures...

File: recipes/recipe-logic-errors.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+*Inaccurate recipe that moves billed sales orders in NetSuite to FinancialForce as invoices without logic to prevent duplicates*
+
+In this case, if a sales order was updated 3 times while it has a status of `Billed`, the recipe will register 3 jobs with this particular sales order, and would have created 3 corresponding FinancialForce invoices. The recipe logic should take into account the possibility of multiple trigger events for the same record, so as to prevent data duplication.
+
+The following recipe demonstrates one way to prevent such data duplication by ensuring that NetSuite sales orders are only moved to FinancialForce once, when they are first marked as `Billed`. It updates NetSuite sales orders with an external ID (with the value of the corresponding FinancialForce's invoice ID), and has an additional trigger condition to ensure that we don't process sales orders with an external ID again.
+
+![Recipe that moves sales orders in NetSuite to FinancialForce invoices the first time status is changed to billed](/assets/images/troubleshooting/recipe-with-trigger-condition-logic.png)
```
Comment (zweedao) [inline recipes/recipe-logic-errors.md]: A bit lost here. In the screenshot, perhaps action 2 should be expanded to clearly show NetSuite sales orders have been updated with external ID. 

Also maybe you want to draw to highlight Trigger line 3 and Action 2, to highlight that they are the de-dup mechanism.

File: recipes/recipe-logic-errors.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+*Recipe that moves sales orders in NetSuite to FinancialForce invoices the first time status is changed to billed*
+
+The following recipe demonstrates another way to prevent such data duplication, by ensuring that NetSuite sales orders are only created once, and subsequently are updated. It creates a new FinancialForce invoice if the NetSuite sales order has no external ID, and updates the corresponding FinancialForce invoice if there is an external ID.
+
+![Example recipe that syncs NetSuite sales orders with FinancialForce invoices](/assets/images/troubleshooting/recipe-with-update-logic.png)
```
Comment (zweedao) [inline recipes/recipe-logic-errors.md]: would be nice too, to have more drawing in this screenshot to clearly show where is the de-dup mechanism

**Comment by huilinn** [comment]
Fixed PR comments, @zweedao can you review?

**Pull request merged**